rest_framework/pagination.py

```diff
@@ -275,9 +275,16 @@ def get_page_size(self, request):
     def get_next_link(self):
         if not self.page.has_next():
             return None
-        url = self.request.build_absolute_uri()
```

Comment:
This calls [comes from Django](https://docs.djangoproject.com/en/5.0/ref/request-response/#django.http.HttpRequest.build_absolute_uri) so if you're getting `http` returned, you probably have a misconfiguration somewhere:

> Mixing HTTP and HTTPS on the same site is discouraged, therefore [build_absolute_uri()](https://docs.djangoproject.com/en/5.0/ref/request-response/#django.http.HttpRequest.build_absolute_uri) will always generate an absolute URI with the same scheme the current request has. If you need to redirect users to HTTPS, it’s best to let your web server redirect all HTTP traffic to HTTPS.

Looking at [the code](https://github.com/django/django/blob/9cb8baa0c4fa2c10789c5c8b65f4465932d4d172/django/http/request.py#L204) for this method, the scheme is computed here: https://github.com/django/django/blob/9cb8baa0c4fa2c10789c5c8b65f4465932d4d172/django/http/request.py#L255-L268

You may want to try setting [`SECURE_PROXY_SSL_HEADER`](https://docs.djangoproject.com/en/stable/ref/settings/#secure-proxy-ssl-header) on the environments where you expect https to be used:

```
SECURE_PROXY_SSL_HEADER = ("HTTP_X_FORWARDED_PROTO", "https")
```